tools/circt-test/circt-test-runner-circt-bmc.py

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env python3
+from pathlib import Path
+import argparse
+import shlex
+import subprocess
+import sys
+import tempfile
+
+parser = argparse.ArgumentParser()
+parser.add_argument("mlir")
+parser.add_argument("-t", "--test", required=True)
+parser.add_argument("-d", "--directory", required=True)
+parser.add_argument("-m", "--mode", type=int)
+parser.add_argument("-k", "--depth")
+args = parser.parse_args()
+
+if args.mode is not None and args.mode != "bmc":
+  sys.stderr.write("error: circt-bmc only supports `bmc` mode")
+
+# Use circt-opt to lower any `verif.formal` ops to `hw.module`s. Once circt-bmc
+# natively supports `verif.formal`, we can directly run it on the input MLIR.
```

Comment:
I think once we have this connection, we'd ideally just instantiate the circt-bmc pipeline and jit exec call in circt-test directly instead of going through a python script.

Reply:
That's definitely an option. Although I'm also a bit hesitant to conflate test discovery and scheduling with the actual implementation of a formal verification tool. Maybe something like `circt-formal` or `circt-verify` could implement the collection of LEC, BMC, induction, and other proofs, and `circt-test` would just call out to that tool? That way we could get rid of the runner in favor of just calling the tools directly. But a separation may be valuable regardless.